xoa_driver/functions/anlt.py

```diff
@@ -247,217 +314,206 @@ async def an_log(port: GenericAnyPort) -> str:
+        md, ps, tm = (LinkTrainingMode.STANDALONE, NRZPreset.NRZ_WITH_PRESET, TimeoutMode.DISABLED)
+    elif t == ('autocomplete', False, True):
+        md, ps, tm = (LinkTrainingMode.STANDALONE, NRZPreset.NRZ_NO_PRESET, TimeoutMode.DEFAULT)
+    elif t == ('autocomplete', False, False):
+        md, ps, tm = (LinkTrainingMode.STANDALONE, NRZPreset.NRZ_NO_PRESET, TimeoutMode.DISABLED)
+
+    elif t == ('disable', True, True):
+        md, ps, tm = (LinkTrainingMode.DISABLE, NRZPreset.NRZ_WITH_PRESET, TimeoutMode.DEFAULT)
+    elif t == ('disable', True, False):
+        md, ps, tm = (LinkTrainingMode.DISABLE, NRZPreset.NRZ_WITH_PRESET, TimeoutMode.DISABLED)
+    elif t == ('disable', False, True):
+        md, ps, tm = (LinkTrainingMode.DISABLE, NRZPreset.NRZ_NO_PRESET, TimeoutMode.DEFAULT)
+    elif t == ('disable', False, False):
+        md, ps, tm = (LinkTrainingMode.DISABLE, NRZPreset.NRZ_NO_PRESET, TimeoutMode.DISABLED)
+
+    elif t == ('interactive', True, True):
+        md, ps, tm = (LinkTrainingMode.INTERACTIVE, NRZPreset.NRZ_WITH_PRESET, TimeoutMode.DEFAULT)
+    elif t == ('interactive', True, False):
+        md, ps, tm = (LinkTrainingMode.INTERACTIVE, NRZPreset.NRZ_WITH_PRESET, TimeoutMode.DISABLED)
+    elif t == ('interactive', False, True):
+        md, ps, tm = (LinkTrainingMode.INTERACTIVE, NRZPreset.NRZ_NO_PRESET, TimeoutMode.DEFAULT)
+    elif t == ('interactive', False, False):
+        md, ps, tm = (LinkTrainingMode.INTERACTIVE, NRZPreset.NRZ_NO_PRESET, TimeoutMode.DISABLED)
+
+    elif t == ('autostart', True, True):
+        md, ps, tm = (LinkTrainingMode.START_AFTER_AUTONEG, NRZPreset.NRZ_WITH_PRESET, TimeoutMode.DEFAULT)
+    elif t == ('autostart', True, False):
+        md, ps, tm = (LinkTrainingMode.START_AFTER_AUTONEG, NRZPreset.NRZ_WITH_PRESET, TimeoutMode.DISABLED)
+    elif t == ('autostart', False, True):
+        md, ps, tm = (LinkTrainingMode.START_AFTER_AUTONEG, NRZPreset.NRZ_NO_PRESET, TimeoutMode.DEFAULT)
+    else:
+        md, ps, tm = (LinkTrainingMode.START_AFTER_AUTONEG, NRZPreset.NRZ_NO_PRESET, TimeoutMode.DISABLED)
 
     tokens += [
         commands.PP_LINKTRAIN(conn, mid, pid).set(
-            mode=lm,
+            mode=LinkTrainingMode.DISABLE,
+            pam4_frame_size=PAM4FrameSize.P16K_FRAME,
+            nrz_pam4_init_cond=LinkTrainingInitCondition.NO_INIT,
+            nrz_preset=ps,
+            timeout_mode=tm,
+        ),
+        commands.PP_LINKTRAIN(conn, mid, pid).set(
+            mode=md,
             pam4_frame_size=PAM4FrameSize.P16K_FRAME,
             nrz_pam4_init_cond=LinkTrainingInitCondition.NO_INIT,
-            nrz_preset=NRZPreset.NRZ_NO_PRESET,
+            nrz_preset=ps,
             timeout_mode=tm,
         ),
     ]
-
-    if link_recovery:
-        page_xindex = 0
-        serdes_xindex = 0
-        rw = commands.PL1_CFG_TMP(conn, mid, pid, page_xindex, serdes_xindex)
-        tokens.append(rw.set(link_recovery))
     await apply(*tokens)
```

Comment:
in this context is no any reason in using of tokens: list, just to pass methods directly to the apply function

can be:
``` python
lnkt = commands.PP_LINKTRAIN(conn, mid, pid)
await apply(
    lnkt.set(
        mode=LinkTrainingMode.DISABLE,
        pam4_frame_size=PAM4FrameSize.P16K_FRAME,
        nrz_pam4_init_cond=LinkTrainingInitCondition.NO_INIT,
        nrz_preset=ps,
        timeout_mode=tm,
    ),
    lnkt.set(
        mode=md,
        pam4_frame_size=PAM4FrameSize.P16K_FRAME,
        nrz_pam4_init_cond=LinkTrainingInitCondition.NO_INIT,
        nrz_preset=ps,
        timeout_mode=tm,
    ),
)
```